dataland-frontend/tests/e2e/specs/eu-taxonomy-financials/EditButtonWithMultipleReportingPeriods.ts

```diff
@@ -0,0 +1,85 @@
+import { DataTypeEnum, EuTaxonomyDataForFinancials, StoredCompany } from "@clients/backend";
+import { describeIf } from "@e2e/support/TestUtility";
+import { getKeycloakToken } from "@e2e/utils/Auth";
+import { generateDummyCompanyInformation, prepareUniqueCompany, uploadCompanyViaApi } from "@e2e/utils/CompanyUpload";
+import { admin_name, admin_pw } from "@e2e/utils/Cypress";
+import { addCreditInstitutionDataset, prepareFixture } from "@e2e/utils/EuTaxonomyFinancialsUpload";
+import { FixtureData, getPreparedFixture } from "@sharedUtils/Fixtures";
+
+describeIf(
+  "As a user, I expect to be able to edit datasets with multiple reporting periods",
+  {
+    executionEnvironments: ["developmentLocal", "ci", "developmentCd"],
+    dataEnvironments: ["fakeFixtures"],
+  },
+  function () {
+    let testData: FixtureData<EuTaxonomyDataForFinancials>;
+    const testCompany = prepareUniqueCompany("company-for-testing-edit-button");
+
+    before(function () {
+      const fixtureType = "CompanyInformationWithEuTaxonomyDataForFinancialsPreparedFixtures";
+      prepareFixture<EuTaxonomyDataForFinancials>(fixtureType).then((preparedFixtures) => {
+        testData = getPreparedFixture("company-for-all-types", preparedFixtures);
+      });
+    });
+
+    it("Check whether newly added dataset has Pending status and can be approved by a reviewer", () => {
+      getKeycloakToken(admin_name, admin_pw).then((token: string) => {
+        return uploadCompanyViaApi(token, generateDummyCompanyInformation(testCompany.companyName)).then(
+          (storedCompany): void => {
+            cy.ensureLoggedIn(admin_name, admin_pw);
+            cy.visitAndCheckAppMount(
+              `/companies/${storedCompany.companyId}/frameworks/${DataTypeEnum.EutaxonomyFinancials}/upload`
+            );
+            addTwoDatasetsWithDifferentReportingPeriods(testData.t, storedCompany);
```

Comment:
Do this via `uploadOneLksgDatasetViaApi()`.  This is way faster then using the forms.
Currently the test takes several minutes on my local machine. This will decrease a lot after you switch.

Reply:
Done.